src/ai/defaultai.cc

```diff
@@ -83,15 +83,6 @@ constexpr int kMineTypes = 4;
 namespace AI {
 
 // Fix undefined references
```

Comment:
I am not sure whether the solution for these values would be so easy. I think solving this issue would need more thoughts as the comment implies that without these declarations the compiler would complaion about missing references. In the header file these values are declared as private members of the defaultAI struct, but maybe they are not always used only in a private context.

Reply:
The compiler warning here says
```
warning: out-of-line definition of constexpr static data member is redundant in C++17 and is deprecated [-Wdeprecated]
```
C++14 and older needs this, C++17 is smarter and makes this superfluous. And apparently it compiles and links fine without them.

Reply:
ok. did not know that. I just ran into issues with these definitions a lot of times, so I was alerted. So I am glad that c++17 is better in this.